src/wp-includes/post.php

```diff
@@ -8018,3 +8019,37 @@ function use_block_editor_for_post_type( $post_type ) {
 	 */
 	return apply_filters( 'use_block_editor_for_post_type', true, $post_type );
 }
+
+/**
+ * Registers any additional post meta fields.
+ *
+ * @since 6.3.0 Adds sync_status meta field to the wp_block post type so an unsynced option can be added.
+ *
+ * @link https://github.com/WordPress/gutenberg/pull/51144
+ *
+ */
+function wp_register_initial_post_meta() {
+	$post_type = 'wp_block';
+	register_post_meta(
+		$post_type,
+		'sync_status',
+		array(
+			'auth_callback'     => function() {
+				return current_user_can( 'edit_posts' );
+			},
+			'sanitize_callback' => 'sanitize_text_field',
+			'single'            => true,
+			'type'              => 'string',
+			'show_in_rest'      => array(
+				'schema' => array(
+					'type'       => 'string',
+					'properties' => array(
+						'sync_status' => array(
+							'type' => 'string',
+						),
+					),
+				),
+			),
```

Comment:
```suggestion
			'show_in_rest'      => array(
				'schema' => array(
					'type'       => 'string',
					'enum' => array( 'fully', 'unsynced' ),
				),
			),
```

This schema entry is incorrect. It's using the `properties` keyword which would only be used when registering a meta key with a `type` of `object`.

We should add an `enum` keyword that describes the possible values the meta field can contain.

Reply:
Thanks. It was initially set as an object, but we reverted to a string, but didn't update the schema correctly. I have updated and set the enum to `array( 'partial', 'unsynced' )` was at this point there won't be a `fully` option as the fully synced blocks don't set the post meta in order to maintain backwards compat with existing reusable blocks. We will be adding a `partial` option post 6.3 though. Should we add that now, or should I just leave the enum as `array( 'unsynced' )` for now?

Reply:
> We will be adding a partial option post 6.3 though. Should we add that now, or should I just leave the enum as array( 'unsynced' ) for now?

No, we shouldn't add it until it becomes available as an option.

> at this point there won't be a fully option as the fully synced blocks don't set the post meta in order to maintain backwards compat with existing reusable blocks.

Have ya'll considered setting the `default` to `fully` in `register_post_meta`? What does the client send when they want to set the reusable block to the equivalent of "fully"?